shibata revised for #37, 38, 39,40 (#42)
fujitake/vantiq-related: vantiq-apps-development/1-day-workshop/docs/eng/troubleshootings.md

## Troubleshooting

> This is a document about frequently encountered problems and how to deal with them.  

Reviewer: ```suggestion
This article explains the frequently encountered problems and how to troubleshoot them.  
```

### 1. **Application does not work.**

> In Vantiq, there are settings of active status/inactive status for some resources such as Source and Application. If it is set to inactive, it will not work. In that case, please change it to active.  

Reviewer: ```suggestion
In Vantiq, some resources such as Source and Application have the status of `active` or `inactive`.  They don't work when set to inactive. Please change them to active.  
```

### 2. **Data cannot be delivered from the Data Generator to the MQTT broker, Data cannot be received by Source.**

> If the Topic name is wrong, it means that you are publishing or subscribing to a Topic not expected. Please confirm the followings: `not include spaces before or after the Topic name`, `capitalization is correct`, `no typos`.

Reviewer: ```suggestion
If the Topic name is wrong, then you are publishing or subscribing to an unintended Topic. Please note: `No spaces before or after the Topic name`, `Topic name is case sensitive`, `no typos`.
```

### 3. **During developing with App Builder, an error occurs.**

Do a "View Task Events" on the task before the task where the error occurs and confirm that the output data is as intended. Also, trace the cause of the error along with the content of the displayed error, and fix it.

### 4. **How to restart the Generator**

> Confirm that the Source is set to active. Click "Launch" > "Run currentry saved Client in Client Launcher (RTC)" in "TrainingDataGeneratorClient" to open the Data Generator in a browser. Click the _Start Generator_ button. Data generation will start.  

Reviewer: ```suggestion
Confirm that the Source is set to active. Click "Launch" > "Run currentry saved Client in Client Launcher (RTC)" in "TrainingDataGeneratorClient" to open the Data Generator in a browser. Click the _Start Generator_ button. Data generation will restart.  
```
